openslides_backend/action/actions/meeting/archive.py

```diff
@@ -22,6 +23,41 @@ class MeetingArchive(UpdateAction, GetMeetingIdFromIdMixin):
     model = Meeting()
     schema = DefaultSchema(Meeting()).get_update_schema()
 
+    def validate_instance(self, instance: dict[str, Any]) -> None:
+        super().validate_instance(instance)
+
+        messages = []
+
+        active_speaker_exists = self.datastore.exists(
+            "speaker",
+            And(
+                FilterOperator("meeting_id", "=", instance["id"]),
+                FilterOperator("begin_time", "!=", None),
+                FilterOperator("end_time", "=", None),
+            ),
+            lock_result=False,
+        )
+        active_polls = self.datastore.filter(
+            "poll",
+            And(
+                FilterOperator("meeting_id", "=", instance["id"]),
+                FilterOperator("state", "=", "started"),
+            ),
+            ["title"],
+            lock_result=False,
+        )
+
+        if active_speaker_exists:
+            messages.append("speaker")
+        if len(active_polls):
+            messages.append(
+                f"polls ({', '.join([model['title'] for model in active_polls.values()])})"
+            )
```

Comment:
If you want to list the specific models, I'd say using the model ids would be better since names can be non-unique (and it's also how we usually do things if it needs to be this specific), but in this case (since the state of all polls in the meeting can actually be viewed in the client) it'd probably be enough to just say `"Cannot archive meeting with active polls."`